### ISSUE TITLE
ACF Date field: Language locale (site setting and locale attribute) not working for logged-out users

### ISSUE THREAD

**Issue by ghen-arvesu** [TangibleInc/template-system#70]
Hey @eliot-akira 

When the user (admin) is logged in, the date display’s correct, in a different languafe. But, the visitors who are not logged in, see the date in English.

The Date is displayed by a ACF date field (datepicker)

```
<Field acf_date=periode_van /> 
<If acf_true_false=periode_tot> t/m <Field acf_date=periode_tot /><Else /><!-- empty --></If>
```

![image](https://github.com/TangibleInc/template-system/assets/81432112/5015e41c-1142-472b-856f-7f31b45cd1ad)

Adding a locale to the Field tag does not seem to work. 

Discourse link: https://discourse.tangible.one/t/different-date-display-when-logged-out/1196

**Comment by BenTangible** [comment]
> Adding a locale to the Field tag does not seem to work.

To expand on that here's a test I did showing that even though the docs mention this should work, the `locale` attribute doesn't seem to do anything on ACF date fields:

![image](https://github.com/TangibleInc/template-system/assets/113368575/918c63eb-9266-49cf-9825-79fe48e2f3d2)

More generally, I also noticed that if I set the site's language to French, for example, the ACF interface in my post meta seems to automatically switch to displaying languages using the French locale/language, as you can see at the bottom of this screenshot. I don't know what ACF's default output is in this context, but I do know that, as shown in the preview in the post content, L&L displays the date in English, again regardless of the site's settings and whether a locale attribute is set.

![image](https://github.com/TangibleInc/template-system/assets/113368575/fd59fecf-8867-4e76-93a7-c1e85d2f7af7)

I hope this info helps replicate the underlying issue. The fact that the forum user Ghen mentioned is seeing something different when logged in/out might be an additional issue on top of the underlying issue shown in my screenshots.

**Comment by eliot-akira** [comment]
Thanks for the issue description and screenshots. It looks like:

- **ACF Date field** is not applying the **site settings for language locale**, maybe due to a field setting.
- `Field` tag's `locale` attribute is not getting passed to the ACF function `get_field()`. Or, it's getting ignored.
- Somehow the current user's logged-in state is affecting this too.

Probably the solution will need to be implemented in this file: https://github.com/TangibleInc/template-system/blob/main/language/tags/field/acf.php

**Comment by eliot-akira** [comment]
From the documentation:

> Optionally, use the `format` and `locale` attributes to apply date formatting.

This sounds like both attributes are needed, not just `locale`. In the next line, it says:

> Use `format=default` to use the site setting from Settings -> General -> Date Format.

This might be what's necessary to make the ACF Date field work as expected. I'm not sure why it's not the default behavior to use site settings for locale, but this could be an intentional design in ACF.

**Comment by GabrielGallagher** [comment]
Should we stealthily add format=default if no format is specified to reduce confusion then?

**Comment by eliot-akira** [comment]
That sounds like a reasonable solution. Though I wonder why it's not already the default - it could be that I'd just overlooked it, or wasn't thinking ahead. I'll check inside ACF to see what they intended, and compare with how it behaves when adding `format=default` by default. From what everyone has described so far, doing so will probably be the most intuitive and reliable result.

**Comment by BenTangible** [comment]
I just did a quick test and whether I do `<Field acf_date="date_field" format="l j F Y" locale="fr" />` or `<Field acf_date="date_field" format="default" locale="fr" />`, it always shows up in English.

Also, if I change the language of the site to French and use ACF's own shortcode `[acf field="date_field" post_id="123"]`, the date is output in French. So ultimately, it shouldn't be necessary to use `format="default" locale="fr"` on the `acf_date` field since the field should be displaying the date in the site's language by default.

**Comment by BenTangible** [comment]
I just tested this and it looks like the correct language is being displayed by default and the `locale` attribute works to optionally modify it. However, I'm reopening this instead of creating a new issue because as far as I can see it's not completely resolved since the `format` attribute still isn't doing anything.

For example, both `<Field acf_date="date_field" format="Y-m-d" locale="fr" />` and <Field acf_date="date_field" format="l j F Y" />`  output "janvier 18, 2024".

**Comment by eliot-akira** [comment]
Ah, good catch, thanks! I've added tests to ensure all combinations of attributes `format`, `locale`, and site settings work together as expected.

**Comment by BenTangible** [comment]
Hi @eliot-akira, sorry for reopening this again, but I think our implementation here had a slight flaw that I just noticed when taking a look at the code in one of your recent commits (thanks for all of those by the way, awesome to see so much progress on a bunch of these issues).

We established above that this was an issue:

> **ACF Date field** is not applying the **site settings for language locale**

This has been resolved, but I just noticed that the implementation included a minor faulty assumption that I noticed in [this commit](https://github.com/TangibleInc/template-system/commit/844422f32cc1c060e64a3d0111b444b2651d9c13):

> Date field: Set default format from Settings -> General -> Date Format

I'm pretty sure that only the language should be informed by the site settings and that for the format, the `acf_date` field should work the same way as the `acf_date_time` and `acf_time` fields. In other words, for all of these field types, L&L should:
1. Use the format specified in ACF
2. Use the language specified in the site settings (which I believe is already used by ACF in the first place so maybe we can just rely on that)
3. Optionally use a different format or language based on the `format` and `language` attributes

Those second and third points are already implemented, but we'll need to fix the `acf_date` field so that it displays the "return format " specified in the ACF settings like the other field types:

![image](https://github.com/TangibleInc/template-system/assets/113368575/51e29edd-6cbc-4f36-9e25-065d88504076)
